pages/_app.tsx

```diff
@@ -3,21 +3,31 @@ import {IntlProvider} from "react-intl";
 
 import messagesRU from '../locales/ru.json'
 import messagesEN from '../locales/en.json'
+import messagesUA from '../locales/uk.json'
 
 import "./style.css"
-import {useEffect, useMemo, useState} from "react";
+import React, {useEffect, useMemo, useState} from "react";
+import LanguageDropdown from "../components/LanguageDropdown";
 
+const LANGUAGES = [{value: 'uk', file: messagesUA}, {value: 'ru', file:messagesRU}, {value: 'en', file: messagesEN}];
```

Comment:
We could change it to a Map/object, so all operations would be simpler and faster:

```typescript
const LANGUAGES = [
  uk: messagesUA,
  ru: messagesRU,
  en: messagesEN,
];
```

Then the `useMemo` would be just:

```typescript
const messages = useMemo(() => LANGUAGES[locale] || LANGUAGES.en, [locale]);
```

And the language auto-detection:

```typescript
const detectedLocale = Object.keys(LANGUAGES).includes(navigatorLanguage) ? navigatorLanguage : 'en';
setLocale(detectedLocale);
```